src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,74 @@
+package duke;
+
+/** Parses user input into command + args. */
+public class Parser {
+
+    public static ParsedCommand parse(String line) {
+        String[] firstSplit = line.trim().split("\\s+", 2);
+        String cmd = firstSplit[0].toLowerCase();
+        String args = firstSplit.length > 1 ? firstSplit[1] : "";
+
+        switch (cmd) {
+            case "bye": return new ParsedCommand(CommandType.BYE, "");
+            case "list": return new ParsedCommand(CommandType.LIST, "");
+            case "todo":
+                if (args.isBlank()) throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+                return new ParsedCommand(CommandType.TODO, args);
```

Comment:
I like how you validate arguments close to where the command is parsed.
Since the “blank args” check repeats for several commands, would extracting a small helper improve readability and reduce duplication? Something like:

```suggestion
    private static String requireArgs(String args, String hint) {
        if (args.isBlank()) throw new DukeException(hint);
        return args;
    }

```
Then each case becomes a one-liner:
```suggestion
    case "todo":    return new ParsedCommand(TODO, requireArgs(args, "OOPS!!! The description of a todo cannot be empty."));
```